src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,80 @@
+package duke;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Handles the reading, writing and saving of duke.txt file
+ */
+public class Storage {
+
+    private String filePath;
+    private String directoryPath;
+
+    /**
+     * Constructor for Storage
+     * @param filePath String containing file path where duke.txt is stored
+     * @param directoryPath String containing directory path where duke.txt is stored
+     */
+    public Storage(String filePath, String directoryPath) {
+        this.filePath = filePath;
+        this.directoryPath = directoryPath;
+    }
+
+    /**
+     * Reads the duke.txt file and saves the saved tasks into a task list
+     * @return Task list containing saved tasks
+     * @throws NeroException Throws when file is not found
+     */
+    TaskList<Task> readFile() throws NeroException {
+        TaskList<Task> taskList = new TaskList<Task>();
+        File directory = new File(directoryPath);
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
+        try {
+            File file = new File(filePath);
+            Scanner sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                String[] cleanedInputs = line.split(" \\| ");
+                boolean isDone = (cleanedInputs[1].equals("0")) ? false : true;
+                if (cleanedInputs[0].equals("T")) {
```

Comment:
Would be good to assign this to a variable to improve readability.

Reply:
Sure, will add that in!